ravenframework/Simulation.py

```diff
@@ -771,37 +820,41 @@ def initiateStep(self, stepName):
 
   def executeStep(self, stepInputDict, stepInstance):
     """
-      This method is aimed to execute and finalize the step
-      (it can be splitted in 2 in case needed)
+      This method executes and finalizes the step
       @ In, stepInputDict, dict, the step dictionary
       @ In, stepInstance, instance, step instance
       @ Out, None
     """
     stepInstance.takeAstep(stepInputDict)
-    #---------------here what is going on? Please add comments-----------------
+    # FilePrint has a finalize method that should be run at the end of the step,
+    # run this method for each instance
     for output in stepInputDict['Output']:
-      if "finalize" in dir(output):
-        output.finalize()
-    self.raiseAMessage('-'*2+' End step {0:50} '.format(stepInstance.name+' of type: '+stepInstance.type)+2*'-'+'\n')#,color='green')
+      if "_printer" in output.__dict__:
+        output._printer.finalize()
```

Comment:
The original code was not finding the "finalize" method because it is inside the _printer object, so it was not performing as expected. The only "finalize" method I found in the code is in `FilePrint.py`. Do we anticipate other outputs with a "finalize" method? If so, I could create a "finalize" method in `PrintEntity.py` that calls the "finalize" method from `FilePrint.py` and go back to the original structure. This would remove the check on the protected object.

Reply:
oh, interesting. I think I agree with your assessment, the Entities should have the finalize method, and in those methods they should call finalize on their Interfaces. Given that this is only implemented for one kind of Entity right now, maybe it's not worth making a system out of it.